test/observable/test_variance.py

```diff
@@ -0,0 +1,129 @@
+import netket as nk
+import netket.experimental as nkx
+import numpy as np
+
+import pytest
+
+from ..variational.finite_diff import central_diff_grad, same_derivatives
+
+
+def _setup(useExactSampler=True):
+    N = 3
+    hi = nk.hilbert.Spin(0.5, N)
+
+    ma = nk.models.RBM(alpha=1)
+    n_samples = 8192
+
+    if useExactSampler:
+        sa = nk.sampler.ExactSampler(hilbert=hi)
+        vs = nk.vqs.MCState(
+            sampler=sa,
+            model=ma,
+            n_samples=n_samples,
+        )
+
+    else:
+        n_discard_per_chain = 1e3
+
+        sa = nk.sampler.MetropolisLocal(hilbert=hi, n_chains_per_rank_or_device=16)
```

Comment:
I think something changed, because this should be n?chains per rank
```suggestion
        sa = nk.sampler.MetropolisLocal(hilbert=hi, n_chains_per_rank=16)
```

and same below